src/python/dxpy/workflow_builder.py

```diff
@@ -69,6 +69,46 @@ def _parse_executable_spec(src_dir, json_file_name, parser):
             raise WorkflowBuilderException("Could not parse {} file as JSON: {}".format(json_file_name, e.args))
 
 
+def _fetch_spec_from_workflow(args, parser):
+    try:
+        source_workflow = dxpy.DXWorkflow(args._from)
+        json_spec = source_workflow.describe(fields={"properties", "details"}, default_fields=True)
+        json_spec = _cleanup_empty_keys(json_spec)
+
+    except Exception as e:
+        raise WorkflowBuilderException("Could not get specs from given workflow {}: {}".format(args._from, e.args))
+
+    return json_spec
+
+def _cleanup_empty_keys(json_spec):
+    import re
+    clean_json = re.sub('\"\w*\": (null|\{\}|\"\"|\[\])(\,|)\s*','',json.dumps(json_spec)).replace(", }","}")
+    return json.loads(clean_json)
+
+def _check_dxcompiler_version(json_spec):
+    if  json_spec.get("details") and json_spec["details"].get("version"):
+        from distutils.version import StrictVersion
+        supported_version = "2.5.0"
+        current_compiler_version = str.split(json_spec["details"].get("version"),"-")[0]
+        if StrictVersion(current_compiler_version) < StrictVersion(supported_version):
+            raise WorkflowBuilderException("Source workflow {} is not compiled using dxCompiler (version>={}) that supports creating global workflows.".format(json_spec["id"], supported_version))
+    else:
+        raise WorkflowBuilderException("Cannot find the dxCompiler version from the spec of the source workflow {}. Please specify the dxCompiler version in the 'details' field of the workflow spec.".format(json_spec["id"]))
+
+def _notify_instance_type_selection(json_spec):
+    is_static = json_spec["details"].get("staticInstanceTypeSelection", False)
+    if is_static:
+        print("Note: {workflow} was compiled with -instanceTypeSelection=static "
+              "and will produce a global workflow that relies on instance types that may not be available to all users."
+              .format(workflow=json_spec["id"]))
+
+def _notify_dependencies(json_spec):
+     dependency_list = json_spec.get(["description"])
+     if dependency_list:
+         print("Note: {workflow} was compiled with unbundled dependencies. "
+              "Please check the workflow description and make sure access to these dependencies is provided to all authorized users."
+              .format(workflow=json_spec["id"]))
```

Comment:
@r-i-v-a @emiloslavsky as said here:https://confluence.internal.dnanexus.com/display/TTN/Global+WDL+workflows

> If we detect a dxCompiler-built local workflow with dependencies and --brief is not specified, notify the global workflow builder to check workflow description for these dependencies and share them with authorized users

So shall I just put a short message like above, or parse the description and print the whole list out?

Reply:
> Updated "details" (if static instance type) looks like
>     "details": {
>         "staticInstanceTypeSelection": true,
>     },
> If dynamic instance type selection, that field will not be in "details."

Reply:
Just put a short message that tells them to check the metadata details field. @YuxinShi0423